newIDE/app/src/ObjectEditor/Editors/SpineEditor.js

```diff
@@ -0,0 +1,528 @@
+}: EditorProps) => {
+  const scrollView = React.useRef<?ScrollViewInterface>(null);
+
+  const getResource = React.useCallback(
+    (name: string) => {
+      const resourcesManager = project.getResourcesManager();
+
+      return resourcesManager.hasResource(name)
+        ? resourcesManager.getResource(name)
+        : null;
+    },
+    [project]
+  );
+  const getMetadata = resource => {
+    const metadataString = resource ? resource.getMetadata() : '';
+
+    return !!metadataString ? JSON.parse(metadataString) : {};
+  };
+  const [
+    justAddedAnimationName,
+    setJustAddedAnimationName,
+  ] = React.useState<?string>(null);
+  const justAddedAnimationElement = React.useRef<?any>(null);
+
+  React.useEffect(
+    () => {
+      if (
+        scrollView.current &&
+        justAddedAnimationElement.current &&
+        justAddedAnimationName
+      ) {
+        scrollView.current.scrollTo(justAddedAnimationElement.current);
+        setJustAddedAnimationName(null);
+        justAddedAnimationElement.current = null;
+      }
+    },
+    [justAddedAnimationName]
+  );
+  const { showAlert } = useAlertDialog();
+
+  const draggedAnimationIndex = React.useRef<number | null>(null);
+
+  const gdevelopTheme = React.useContext(GDevelopThemeContext);
+  const forceUpdate = useForceUpdate();
+
+  const spineConfiguration = gd.asSpineConfiguration(objectConfiguration);
+  const properties = objectConfiguration.getProperties();
+
+  const [nameErrors, setNameErrors] = React.useState<{ [number]: React.Node }>(
+    {}
+  );
+
+  const [skeleton, setSkeleton] = React.useState<?ISkeletonData>(null);
+  const getEmbeddedResourcesMapping = React.useCallback(
+    (resourceName: string): { [string]: string } => {
+      const resource = getResource(resourceName);
+
+      return getMetadata(resource).embeddedResourcesMapping;
+    },
+    [getResource]
+  );
+  const getSkeleton = React.useCallback(
+    (spineResourceName: string) => {
+      const jsonResourcesMapping = getEmbeddedResourcesMapping(
+        spineResourceName
+      );
+      if (!jsonResourcesMapping) return Promise.resolve(undefined);
+
+      const jsonResourcesMappingValues = Object.values(jsonResourcesMapping);
+      const textureAtlasName = jsonResourcesMappingValues[0];
+
+      // flow check
+      if (typeof textureAtlasName !== 'string')
+        return Promise.resolve(undefined);
+
+      const atlasResourcesMapping = getEmbeddedResourcesMapping(
+        textureAtlasName
+      );
+
+      if (
+        !atlasResourcesMapping ||
+        !Object.values(atlasResourcesMapping).length
+      )
+        return Promise.resolve(undefined);
+
+      return PixiResourcesLoader.getSpineData(project, spineResourceName).then(
+        newSkeleton => {
+          setSkeleton(newSkeleton);
+
+          return newSkeleton;
+        }
+      );
+    },
+    [project, getEmbeddedResourcesMapping]
+  );
+  getSkeleton(properties.get('spineResourceName').getValue());
```

Comment:
Should that line doing getSkeleton() be in a React.useEffect? Having something called in a component like this at each render seems wasteful at best, dangerous at worst?

Using an effect also allow to make it clear that there is a state change implied: just remove the `setSkeleton` and call it manually.

I would also rename `getSkeleton` to `loadSkeleton` (it does something more involved than just getting - even after the side effect `setSkeleton` is removed)

```suggestion
      return PixiResourcesLoader.getSpineData(project, spineResourceName);
    },
    [project, getEmbeddedResourcesMapping]
  );
  React.useEffect(() => {
    (async() => {
      const skeleton = await loadSkeleton(properties.get('spineResourceName').getValue());
      setSkeleton(skeleton);
    })();
  }, []);
```

Reply:
@4ian React is not my good side. Its learning in my TODOs so I hope for your help here for now. 
I updated. Could you please check? In addition I solved all other comments

Thanks

Reply:
Thank you I will check 👍